### PR TITLE
Add interface for depth in both forward rendering and backward propagation

### PR DESCRIPTION
**Pull request by ingra14m**
Test in self-defined dataset with GT-depth
depth-GT
Without depth loss:

With depth loss

### CODE REVIEW

**Comment by XuyangBai** [comment]
Should there be a second part of gradient affected by the depth loss, the dL_dtz in L268 is also affected by the dL_depths?

**Comment by cv-lab-x** [comment]
> Test in self-defined dataset with GT-depth depth-GT Without depth loss: 
> 
> With depth loss 

Hi，thanks for your work, have you transformed the render depths to point3ds? I test your latest branch on datasets bicycle mip360-datasets, it seems the render depths have some erros, and multi views's render depths are not  consistent. Looking forward to your reply, thanks! @ingra14m

**Comment by shippoT** [comment]
Hi，how do you get the depth map shown in ”Test in self-defined dataset with GT-depth“. I rendered the same NeRF Synthetic dataset using blender，but got quite different result.
000_depth0000

**Comment by ingra14m** [comment]
If I remember correctly, first, I set the parts where alpha=0 to be black. 

Second, since the depth directly from Blender shows deeper depths in blacker colors, I used `1.0 - normalized depth`. In this way, the depth gt can be directly used to supervise the depth directly output by gs.

**Comment by arcman7** [comment]
> Hi，thanks for your work, have you transformed the render depths to point3ds? I test your latest branch on datasets bicycle mip360-datasets, it seems the render depths have some erros, and multi views's render depths are not consistent. Looking forward to your reply, thanks! @ingra14m

What's the status of this branch? Does including the depth information improve the splat fitting or rendering?

**Comment by ingra14m** [comment]
Hi @arcman7, from my perspective, I think depth can not improve the rendering quality of 3D Gaussian splatting. The geometry enhancement cannot lead to a better rendering quality.

**Comment by arcman7** [comment]
> Hi @arcman7, from my perspective, I think depth can not improve the rendering quality of 3D Gaussian splatting. The geometry enhancement cannot lead to a better rendering quality.

Ah okay, thanks for the heads up - it was looking really hopeful when I was going through all of the related discussion threads and experiments people had setup

**Comment by arcman7** [comment]
Hey btw, I wanted to ask you a quick question, and I didn't think creating another issue on this busy repo was the way to go; is there a way to render a pretrained gaussian model without setting up training gradient tensors in cuda? I'm not trying to improve or modify this pre-existing gaussian splat model, I just want to rasterize it from it's various camera view points but using the existing py-torch rasterization methods in this repo

**Comment by jingyuzhao1010** [comment]
> > Test in self-defined dataset with GT-depth depth-GT Without depth loss: 
> > With depth loss 
> 
> Hi，thanks for your work, have you transformed the render depths to point3ds? I test your latest branch on datasets bicycle mip360-datasets, it seems the render depths have some erros, and multi views's render depths are not consistent. Looking forward to your reply, thanks! @ingra14m

Hi,
May I ask if the depth problem of this branch has been solved? I also encountered the problem that the multi-frame view cannot be aligned after being projected to 3d.@cv-lab-x

**Comment by jingyuzhao1010** [comment]
> Test in self-defined dataset with GT-depth depth-GT Without depth loss: 
> 
> With depth loss 


Hi,
May I ask why after I use your depth visualization cuda code under this branch, the rendered depth map output is all 255,255,255?
Your prompt reply will be highly appreciated!@ingra14m
